store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreConfiguredGraphIdsTest.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    private static FederatedStore getFederatedStoreFromJson(final String path) throws IOException, StoreException {
+        final FederatedStore federatedStore = loadFederatedStoreFrom(path);
+        federatedStore.initialise(GRAPH_ID_SELECTED, new Schema(), new FederatedStoreProperties());
+        addDefaultLibrary(federatedStore);
+        return federatedStore;
+    }
+
+    private static FederatedStore getFederatedStoreFromProperties(final String path) {
+        final FederatedStore federatedStore = (FederatedStore) Store.createStore(GRAPH_ID_TEST_FEDERATED_STORE,
+                STRING_SCHEMA,
+                StoreProperties.loadStoreProperties(StreamUtil.openStream(FederatedStore.class, path)));
+
+        addDefaultLibrary(federatedStore);
+        return federatedStore;
+    }
+
+    private static void addDefaultLibrary(final FederatedStore federatedStore) {
+        final HashMapGraphLibrary library = new HashMapGraphLibrary();
+        library.addProperties(STORE_PROP_ID, PROPERTIES);
+        federatedStore.setGraphLibrary(library);
+    }
+
+    private static Schema addSelectedSchemaAndGraph(final FederatedStore federatedStore)
+            throws OperationException {
+        final Schema selectedSchema = new Schema.Builder()
+                .edge("edge", new SchemaEdgeDefinition.Builder()
+                        .source(STRING)
+                        .destination(STRING)
+                        .property(PROPERTY_1, STRING)
+                        .directed(DIRECTED_EITHER)
+                        .build())
+                .vertexSerialiser(new StringSerialiser())
+                .type(DIRECTED_EITHER, Boolean.class)
+                .merge(STRING_SCHEMA)
+                .build();
+
+        federatedStore.getGraphLibrary().addSchema(SCHEMA_SELECTED, selectedSchema);
+        addGraph(federatedStore, GRAPH_ID_SELECTED, SCHEMA_SELECTED, contextTestUser());
+
+        return selectedSchema;
+    }
+
+    private static Schema addUnselectedSchemaAndGraph(final FederatedStore federatedStore)
+            throws OperationException {
+        final Schema unselectedSchema = new Schema.Builder()
+                .edge("edge", new SchemaEdgeDefinition.Builder()
+                        .source(STRING)
+                        .destination(STRING)
+                        .property(PROPERTY_2, STRING)
+                        .directed(DIRECTED_EITHER)
+                        .build())
+                .vertexSerialiser(new StringSerialiser())
+                .type(DIRECTED_EITHER, Boolean.class)
+                .merge(STRING_SCHEMA)
+                .build();
+
+        federatedStore.getGraphLibrary().addSchema(SCHEMA_UNSELECTED, unselectedSchema);
+        addGraph(federatedStore, GRAPH_ID_UNSELECTED, SCHEMA_UNSELECTED, contextTestUser());
+
+        return unselectedSchema;
+    }
+
+    private static void addGraph(final FederatedStore federatedStore, final String graphId,
```

Comment:
There's a method in `FederatedStoreTestUtil` very similar to this. If that method doesn't quite work another could be added there instead.

Reply:
The 2 are different in the way they call `AddGraph`. One is a `FederatedSore` direct call and other uses an operation. One presents the schema and properties where as the other stores the schema and properties in the cache and gives a sting representation that is then used later to get the schemas/properties and merge them returning a single schema. So, if you wanted to amalgamate the 2 you would have to convert to a single format and provide more options to the util class. If you were to go down the `Operation` route then it includes adding an exception catch to 2 other methods.

I have gone down the `Operation` route. It functions as before but you have lost the direct call as a result. Is that important to you? I have not merged the changes yet. The new code:

```
public static void addGraphToAccumuloStore(final FederatedStore federatedStore, final String graphId,
                                               final boolean isPublic, final Schema schema)
        throws StorageException, OperationException {
    addGraph(federatedStore, graphId, isPublic,
            loadAccumuloStoreProperties(ACCUMULO_STORE_SINGLE_USE_PROPERTIES),
            schema, null, null, new Context(new User(UNKNOWN_USER_ID)));
}

public static void addGraph(final FederatedStore federatedStore, final String graphId,
                            final boolean isPublic, final StoreProperties storeProperties, final Schema schema,
                            final String parentPropertyIds, final List<String> parentSchemaIds,
                            final Context context)
        throws OperationException {
    federatedStore.execute(OperationChain.wrap(
            new AddGraph.Builder()
                    .graphId(graphId)
                    .storeProperties(storeProperties)
                    .schema(schema)
                    .parentSchemaIds(parentSchemaIds)
                    .parentPropertiesId(parentPropertyIds)
                    .isPublic(isPublic)
                    .build()), context);
}
```

I altered the method name slightly as it doesn't just add the graph.